queue/QueueConsumer.js

```diff
@@ -24,13 +52,57 @@ module.exports = class QueueConsumer extends BaseQueue {
     if (!this._channel) {
       throw new Error('Channel not initialized')
     }
+
+    const messageContent = message.content
+
+    const maxMessageRetries = this._maxMessageRetries
+    const deadLetterQueueName = this._deadLetterQueueName
+
+    const { headers } = message.properties
+    const deadLetterHeader = headers[DEAD_LETTER_HEADER]
+
+    let rejectedRecord
+    if (deadLetterHeader) {
+      // find 'rejected' record
+      // TODO: handle other reasons in the future
+      // ex. dead letter messages from TTL expires (not implemented yet)
+      for (const record of deadLetterHeader) {
+        if (record.reason === 'rejected') {
+          rejectedRecord = record
```

Comment:
I think you should break here. Is it possible that there would be more than one record that's rejected? My guess is no, but if it is possible then how do we handle the count?